### ISSUE TITLE
On press 27 march does not work timezone not supported, daylight saving 

### ISSUE THREAD

**Issue by SwitiSM** [stephy/CalendarPicker#181]
On press 27 march does not select 27 day and the value of selected start date is 26, while pressing on 26 select both 26 and 27, this is only happening on date 27 but am not sure that all other dates are working correctly, this crash the application on release mode and on development mode gives maximum depth exceeded error.  

![Screenshot_1581511380](https://user-images.githubusercontent.com/45052890/74335949-fdba2980-4da5-11ea-82d0-550a8736f149.png)

**Comment by peacechen** [comment]
Sorry for not seeing this earlier.  Please post your `<CalendarPicker>` code.

**Comment by SwitiSM** [comment]
My code is exactly as default Calendar Example but this is my Mobile phone timezone configurations, we have daylight saving and it's on 27 march (Jordan - Amman) that we change the date one hour forward. You may want to try these configurations on your mobile phone or emulator to reproduce this issue. 

![Screenshot_20200324-143942_Settings](https://user-images.githubusercontent.com/45052890/77426697-add77500-6ddd-11ea-8c57-08eae6193316.jpg)

**Comment by peacechen** [comment]
CalendarPicker compares based on day granularity.  This has avoided time boundary issues especially with daylight saving time changes.  What device make & model are you testing on, and what version of Android?

I suspect that the bug could occur if the time zone change happens at midnight.  That would alter the date boundary (midnight on the 27th suddenly becomes 11pm the 26th).  At what specific time does the time zone change occur?  Here in the states the change occurs at 2am which avoids interfering with the date boundary.

**Comment by SwitiSM** [comment]
"_What device make & model are you testing on, and what version of Android_" It doesn't matter the issue appears even on emulators. 
You're correct it do happen exactly at **midnight**. 
I've tried your change "const thisDay = moment({year, month, day, hour: 12});"
selecting 26th of march does not select 27th anymore that's great ! but selecting 27th halts and **crash** the application giving maximum update depth exceeded error
![Screenshot_1585082615](https://user-images.githubusercontent.com/45052890/77475152-054b0480-6e21-11ea-88cb-85a920e42af6.png)
.

**Comment by peacechen** [comment]
I'll need your help debugging this since I don't have your exact setup.  Would you put a breakpoint inside `componentDidUpdate`:
https://github.com/stephy/CalendarPicker/blob/ef7d242afd7083507d9a91a265cb495a86044d4d/CalendarPicker/index.js#L52

Try to determine which prop is changing repeatedly and causing an infinite loop.  If that isn't the culprit, set a breakpoint in Day.js or the other child components to see if they're getting stuck in `componentDidUpdate`.

**Comment by SwitiSM** [comment]
https://github.com/stephy/CalendarPicker/blob/ef7d242afd7083507d9a91a265cb495a86044d4d/CalendarPicker/index.js#L64

This line always yields true on selecting 27th march on release mode or without Debugging JS remotely.  

**prevProps.initialDate** ---> Thu Mar 26 2020 23:00 :00 GMT+0200 (EET)
**moment(prevProps.initialDate)** ---> 1585256400000 _which is Thu Mar 26 2020 23:00:00 GMT+0200 (Eastern European Standard Time) based on JavaScript new Date(1585256400000 )_
**this.props.initialDate** ---> Thu Mar 26 2020 23:00 :00 GMT+0200 (EET)


Please note that using remote Js debug mode does not produce the issue. Note the following : 

[https://momentjscom.readthedocs.io/en/latest/moment/05-query/02-is-same/ ](url)

> NOTE: moment().isSame() **has undefined behavior and should not be used!** If the code runs fast the initial created moment would be the same as the one created in isSame to perform the check, so the result would be true. But if the code runs slower it's possible that the moment created in isSame is measurably after the one created in moment(), so the call would return false.

**Comment by peacechen** [comment]
Thanks for debugging this.  The underlying issue is that the daylight saving time change in your area occurs at midnight and corrupts the date (Mar 27 becomes Mar 26).  CalendarPicker uses what's passed into the `initialDate` prop and has no way to determine whether to shift the date.  Technically it could take into account all the different time zone changes around the world, but that's out of scope for this library and would create a maintenance nightmare (e.g. a government decides to change its dates for DST).

To fix this in your code, pass in a datetime in the middle of the day, similar to how it was constructed with `{hour: 12}` in the change above.  This prevents the date from shifting on the DST boundary.  With this adjustment, the CalendarPicker code may not need to be modified.  Try reverting the change in Day.js line 27.  However there is a chance that all the date comparisons may need to be changed to fully fix this.

Regarding the note on Moment's `isSame` method, that issue occurs when no parameter is passed into moment's constructor, which creates the current datetime.  That causes race condition hence the undefined behavior.

**Comment by peacechen** [comment]
@SwitiSM   I pushed changes that may help your issue.  To pull directly from this repo, use this in your package.json:
```
dependencies: {
    "react-native-calendar-picker": "https://github.com/stephy/CalendarPicker",
}
```

You'll still need to pass in `initialDate` and/or `selectedDate` constructed with `{hour: 12}` to prevent the date from changing on the boundary.

**Comment by SwitiSM** [comment]
Perfect ! It's working , thank you very very very much. 

> You'll still need to pass in initialDate and/or selectedDate constructed with {hour: 12} to prevent the date from changing on the boundary.

Even if I didn't pass the constructor with {hour:12} It does not change the boundary and selects the correct date. Unless I didn't get what you mean by  
> changing on the boundary

**Comment by peacechen** [comment]
That's a better than expected result 👍 

6.0.6 has been published on npm.  For reference, this was fixed in commit 8250844f6119094924625bb97a06b0848371515f
